### ISSUE TITLE
adios_reorganize on ADIOS1 files

### ISSUE THREAD

**Issue by NAThompson** [ornladios/ADIOS2#2189]
I have a set of ADIOS1 files, whose data is in a set of files

```
$ find . -name "xgc.3d*"
./xgc.3d.13500.bp
./xgc.3d.13700.bp
./xgc.3d.13100.bp
./xgc.3d.13800.bp
./xgc.3d.13000.bp
./xgc.3d.13200.bp
./xgc.3d.13400.bp
./xgc.3d.13600.bp
./xgc.3d.13300.bp
./xgc.3d.13900.bp
```

I would like to assemble these into a single BP4 directory which can be read with the `BeginStep()` and `EndStep()` utilities of ADIOS. Can this be achieved with `adios_reorganize`?

**Comment by williamfgc** [comment]
@NAThompson I'd chekc with @jychoi-hpc first, he had a workaround to make those XGC files compatible with adios2 BP3 readers. adios_reorganize can then turn those BP3 datasets into BP4

**Comment by pnorbert** [comment]
adios_reorganize itself does not provide this functionality. We talked about streaming over sets of files for a long time but haven't implemented it in adios yet, see #1046

**Comment by jychoi-hpc** [comment]
@NAThompson Are those files I shared with you yesterday? just for your information, those are Adios2 files.

**Comment by williamfgc** [comment]
@NAThompson yes, I meant handling each file individually, but won't be hard if you write a loop in a script. 

@pnorbert I started exploring that a while ago see [my branch](https://github.com/ornladios/ADIOS2/compare/master...williamfgc:eco_series) the key is the [StreamSeries class](https://github.com/ornladios/ADIOS2/compare/master...williamfgc:eco_series#diff-dedbbab589331837476cc727f77fcc30R40) but it wasn't priority. My impression is that these kind of optional "higher-level" capabilities build on top of the adios2 library (kind of like the bindings, but bindings are mandatory), it could always be a sub-module, plugins, or a separate project to prevent polluting the library as it can increase the size and compilation times pretty quickly. Anyways, I just worked on a project in which we had to stay away from higher-level "hierarchical" access pattern to get better at  performance. Hope it helps.
